### ISSUE TITLE
bug: Crash with Create mod

### ISSUE THREAD

**Issue by Woxedy** [rvbsm/fsit#34]
### 🔍 What happened?

Deployer causes crash if it interacts with player.
If I understand correctly, this mechanism simulates pressing the right mouse button on the player, and thereby tries to sit on it, but obviously it does not work and therefore a crash occurs

### 🤔 What did you expect to happen?

No seat interaction for this block and no game crash.

### 📦 Version

2.1.0

### ⛏ Minecraft Version

1.20.1

### 📃 Relevant log output

```bash
Description: Exception in server tick loop

java.lang.StackOverflowError: Exception in server tick loop
	at net.minecraft.class_1297.addPassengersDeep(class_1297.java:17381)
	at net.minecraft.class_1297.addPassengersDeep(class_1297.java:17385)
	at net.minecraft.class_1297.addPassengersDeep(class_1297.java:17385)
	at net.minecraft.class_1297.addPassengersDeep(class_1297.java:17385)
```

**Comment by rvbsm** [comment]
Wow, I'll take a look into it. Thanks for opening the issue

**Comment by rvbsm** [comment]
The Deployer block uses the player who placed it for interaction, and so the player tries to start riding itself. Since the `DeployerFakePlayer` has a different id than the player, they are different entities, but the interaction is still done as the same player.

I've added a UUID comparison check and can confirm that the game doesn't crash anymore, but I cannot say for sure that the issue is solved.

As far as I understand it, re-login to the world/server after placing the Deployer block can fix the issue until the next version is released.

Remains open until the next release

**Comment by rvbsm** [comment]
Should be fixed in the v2.2.0
